mlrun/utils/notifications/notification_pusher.py

```diff
@@ -118,21 +118,39 @@ def __init__(
         ] = []
 
         for run in self._runs:
-            if isinstance(run, dict):
-                run = mlrun.model.RunObject.from_dict(run)
-
-            for notification in run.spec.notifications:
-                try:
-                    notification.status = run.status.notifications.get(
-                        notification.name
-                    ).get("status", mlrun.common.schemas.NotificationStatus.PENDING)
-                except (AttributeError, KeyError):
-                    notification.status = (
-                        mlrun.common.schemas.NotificationStatus.PENDING
-                    )
-
-                if self._should_notify(run, notification):
-                    self._load_notification(run, notification)
+            try:
+                if isinstance(run, dict):
+                    run = mlrun.model.RunObject.from_dict(run)
+
+                for notification in run.spec.notifications:
+                    try:
+                        try:
```

Comment:
It could be look better I think, find a way to use less try/except and less indentation
overall the idea of not failing on sending a single notificaiton is legit, just need to ensure that code-looking and robustness is not affected by it